sys/auto_init/usb/auto_init_usb.c

```diff
@@ -39,6 +40,51 @@ usbus_cdcecm_device_t cdcecm;
 #include "usb/usbus/dfu.h"
 static usbus_dfu_device_t dfu;
 #endif
+#ifdef MODULE_USBUS_HID
+#include "usb/usbus/hid.h"
+#endif
+#ifdef MODULE_USBUS_MSC
+#include "usb/usbus/msc.h"
+#endif
+
+#ifndef MODULE_USBUS_CDC_ACM
+#define USBUS_CDC_ACM_EP_IN_REQUIRED_NUMOF  0
+#define USBUS_CDC_ACM_EP_OUT_REQUIRED_NUMOF 0
+#endif
+
+#ifndef MODULE_USBUS_CDC_ECM
+#define USBUS_CDC_ECM_EP_IN_REQUIRED_NUMOF  0
+#define USBUS_CDC_ECM_EP_OUT_REQUIRED_NUMOF 0
+#endif
+
+#ifndef MODULE_USBUS_HID
+#define USBUS_HID_EP_IN_REQUIRED_NUMOF      0
+#define USBUS_HID_EP_OUT_REQUIRED_NUMOF     0
+#endif
+
+#ifndef MODULE_USBUS_MSC
+#define USBUS_MSC_EP_IN_REQUIRED_NUMOF      0
+#define USBUS_MSC_EP_OUT_REQUIRED_NUMOF     0
+#endif
+
+#define USBUS_EP_IN_REQUIRED_NUMOF  (USBUS_CONTROL_EP_IN_REQUIRED_NUMOF + \
+                                     USBUS_CDC_ACM_EP_IN_REQUIRED_NUMOF + \
+                                     USBUS_CDC_ECM_EP_IN_REQUIRED_NUMOF + \
+                                     USBUS_HID_EP_IN_REQUIRED_NUMOF + \
+                                     USBUS_MSC_EP_IN_REQUIRED_NUMOF)
+
+#define USBUS_EP_OUT_REQUIRED_NUMOF (USBUS_CONTROL_EP_OUT_REQUIRED_NUMOF + \
+                                     USBUS_CDC_ACM_EP_OUT_REQUIRED_NUMOF + \
+                                     USBUS_CDC_ECM_EP_OUT_REQUIRED_NUMOF + \
+                                     USBUS_HID_EP_OUT_REQUIRED_NUMOF + \
+                                     USBUS_MSC_EP_OUT_REQUIRED_NUMOF)
+
+_Static_assert(USBUS_EP_IN_REQUIRED_NUMOF <= USBDEV_NUM_ENDPOINTS,
```

Comment:
```suggestion
assert(USBUS_EP_IN_REQUIRED_NUMOF <= USBDEV_NUM_ENDPOINTS,
```
For consistency across the file.

Reply:
It is intentionally a `_Static_assert` to check it already during compile time. The problem with `assert` is that it is checked at runtime, which can't be observed if USBUS isn't working and `stdio_cdc_acm` is used. If `assert` were applicable, it would be sufficient to check if `usbdev_new_ep` returns something other than `NULL` (PR #19362 does the check for dynamic configurations during runtime by `assert`).

Reply:
Or did you mean that C11 macro `static_assert` should be used instead of `_Static_assert`?

Reply:
> Or did you mean that C11 macro static_assert should be used instead of _Static_assert?

Yes that's what I was thinking.
Thanks !